lib/frr_zmq.c

```diff
@@ -310,7 +310,7 @@ void frrzmq_thread_cancel(struct frrzmq_cb **cb, struct cb_core *core)
 	thread_cancel(&core->thread);
 
 	if ((*cb)->read.cancelled && !(*cb)->read.thread
-	    && (*cb)->write.cancelled && (*cb)->write.thread)
+	    && (*cb)->write.cancelled && !(*cb)->write.thread)
```

Comment:
seems like this should be two commits?  

a) using thread_add_read/thread_add_write/thread_add_event
b) logic fixes here and in frrmzq_check_events?

Reply:
sure, whatever is the right thing to do: it seemed like it was ok to include this one with the others?